### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by omarbabou**
1. Create a class Classroom with the

  > -   `@label` instance variable that should be initialized in the constructor,
  > -   Setter and getter for `@label `(using `attr_accessor`).

 2. Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
 

  > - Create the `has-many `side (a classroom has many students).
  > - Create the `belongs-to` side (a student belongs to a classroom).
  > - When adding a student to a classroom it also sets the classroom for the student.
  > - When setting the classroom for a student it also adds it to the classrooms' students.

 3. Create a class Book with the following:

  > - `@title` and `@author` instance variables should be initialized in the constructor.
  > - Setters and getters for instance variables (using `attr_accessor`).

 4. Create a class Rental with the following:
 
  > - `@date` instance variable should be initialized in the constructor.
  > - Setter and getter for @Date (remember about `attr_accessor`).

 5. Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class 
 Rental. The following should be implemented:

  > - Create the `has-many` side of Book and Rental (a book has many rentals).
  > - Create the `belongs-to` side of Rental and Book (a rental belongs to a book).
  > - Create the `has-many` side of Person and Rental (a person has many rentals).
  > - Create the `belongs-to` side of Rental and Person (a rental belongs to a person).
  > - Modify the `constructor` of Rental so Book and Person are set in it.

### CODE REVIEW

File: src/rental.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  def initialize(data, book, person)
+    @data = data
```
Comment (julie-ify) [inline src/rental.rb]: - [x] Based on your implementation this should be `date`, not `data`. Kindly fix that.

File: src/book.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Book
+  def initialize(title, author)
+    @title = title,
```
Comment (julie-ify) [inline src/book.rb]: - [x] I don't think this comma is necessary and would throw errors. Kindly remove it.

  ```
  @title = title
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @omarbabou,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @omarbabou,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**